include/lsst/afw/table/python/catalog.h

```diff
@@ -39,11 +39,11 @@ using PyCatalog = pybind11::class_<CatalogT<Record>, std::shared_ptr<CatalogT<Re
 
 /// Extract a column from a potentially non-contiguous Catalog
 template <typename T, typename Record>
-ndarray::Array<typename Field<T>::Value const, 1, 0> _getArrayFromCatalog(
+ndarray::Array<typename Field<T>::Value const, 1, 1> _getArrayFromCatalog(
     CatalogT<Record> const& catalog,  ///< Catalog
     Key<T> const& key  ///< Key to column to extract
 ) {
-    ndarray::Array<typename Field<T>::Value, 1, 0> out = ndarray::allocate(catalog.size());
+    ndarray::Array<typename Field<T>::Value, 1, 1> out = ndarray::allocate(catalog.size());
```

Comment:
How many more of these are out there? And would `auto` work nicely for them to prevent such issues in the future?

Reply:
These are not a big deal; it basically boils down to whether a for loop has `n += s` (where `s == 1`) instead of just `++n`.  I just happened to notice it here.

`auto` won't help at all and in fact would hurt quite a bit, because `ndarray::allocate` doesn't return an `Array` and we're relying on implicit conversions in an expression-template kind of way.